app/models/concerns/user/low_balance_fraud_check.rb

```diff
@@ -29,7 +31,20 @@ def check_for_low_balance_and_probate(refunded_or_disputed_purchase_id)
     disable_refunds_and_put_on_probation! unless recently_probated_for_low_balance?
   end
 
+  def mark_compliant_if_balance_recovered!
+    return unless can_recover_from_low_balance_probation?
+    self.mark_compliant!(author_name: LOW_BALANCE_FRAUD_CHECK_AUTHOR_NAME, content: "Marked compliant automatically on #{Time.current.to_fs(:formatted_date_full_month)} as balance has recovered to #{MoneyFormatter.format(LOW_BALANCE_RECOVERY_THRESHOLD, :usd, no_cents_if_whole: true, symbol: true)}")
+  end
+
   private
+    def can_recover_from_low_balance_probation?
+      return false unless self.on_probation?
+      comments.with_type_on_probation
+              .order(created_at: :desc)
+              .first&.author_name == LOW_BALANCE_FRAUD_CHECK_AUTHOR_NAME &&
```

Comment:
Checking if the latest probation comment is system generated.

This ensures that we don't recover compliance state on probation from admins or other authors